### PR TITLE
Feature/extract sdk

### PR DESCRIPTION
**Pull request by JIexa**
- Removed two types of Id in DID (IROHA and UUID). Added USERNAME type instead. Also, changes have been made in DID-constraint according to last agreement (DID now looks like `did:sora:soraUser`);
- Changed validation process (previously, it was possible  to sign DDO not only by an owner of DDO. Since now this logic has been thrown away);
- Refactored code according to sora-team agreement (but not everywhere);
- Logging to the DB was thrown away too;

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -67,68 +65,41 @@ repositories {
...
 
     //logging
-//    compile("org.apache.logging.log4j:log4j-api:${log4jVersion}")
-//    compile("org.apache.logging.log4j:log4j-core:${log4jVersion}")
-//    compile("org.slf4j:slf4j-api:${slf4jVersion}")
-    implementation("org.aspectj:aspectjrt:${aspectjVersion}")
-    implementation("org.aspectj:aspectjweaver:${aspectjVersion}")
-    compile("com.mchange:c3p0:${c3p0Version}")
+    compile("org.aspectj:aspectjrt:${aspectjVersion}")
+    compile("org.aspectj:aspectjweaver:${aspectjVersion}")
```
Comment (tdorzhi) [inline build.gradle]: we might need `sleuth` and `zipkin` here too

File: src/main/java/jp/co/soramitsu/sora/didresolver/commons/DIDTypeEnum.java [hunk 2]
```diff
@@ -6,8 +6,7 @@
 import lombok.val;
 
 public enum DIDTypeEnum {
-  IROHA("(?<ACCOUNT>\\w+)@(?<DOMAIN>\\w+[\\w|.]*)"),
-  UUID("[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}");
+  USERNAME("[a-zA-Z0-9]{6,32}");
```
Comment (tdorzhi) [inline src/main/java/jp/co/soramitsu/sora/didresolver/commons/DIDTypeEnum.java]: DID spec doesn't have such constraint:
```
idstring           = 1*idchar
idchar             = ALPHA / DIGIT / "." / "-"
```
should we comply with spec in this case, @Warchant ?

File: src/main/java/jp/co/soramitsu/sora/didresolver/controllers/dto/ErrorRs.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package jp.co.soramitsu.sora.didresolver.controllers.dto;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class ErrorRs {
```
Comment (tdorzhi) [inline src/main/java/jp/co/soramitsu/sora/didresolver/controllers/dto/ErrorRs.java]: we may want to add `ApiModel` annotation here, so that swagger-ui sees it

File: src/main/java/jp/co/soramitsu/sora/didresolver/services/VerifyService.java [hunk 4]
```diff
@@ -2,27 +2,35 @@
...
    */
-  Optional<PublicKey> getPublicKeyByProof(Proof proof, List<PublicKey> publicKeys);
+  boolean isCreatorInAuth(DID creator, List<Authentication> authentication);
+
+  /**
+   * Verifies integrity of the DDO
+   *
+   * @param ddo - DDO that is needed to be verified
+   */
+  void verifyIntegrityOfDDO(DDO ddo);
```
Comment (tdorzhi) [inline src/main/java/jp/co/soramitsu/sora/didresolver/services/VerifyService.java]: by name it seems that method should return boolean, but it doesn't

File: src/main/java/jp/co/soramitsu/sora/didresolver/services/impl/VerifyServiceImpl.java [hunk 5]
```diff
@@ -29,44 +35,71 @@
...
+  public boolean isCreatorInPublicKeys(@NotNull DID proofCreator, List<PublicKey> publicKeys) {
+    return publicKeys.stream().anyMatch(key -> proofCreator.equals(key.getId()));
+  }
+
+  @Override
+  public boolean isCreatorInAuth(
+      @NotNull DID creator, @NotNull @Valid List<Authentication> authentication) {
+
+    return true;
+//     FIXME: 11/09/2018 NPE due to incorrect constructing of Authentication
```
[the remark this answers is not in the document]
Comment (JIexa) [reply src/main/java/jp/co/soramitsu/sora/didresolver/services/impl/VerifyServiceImpl.java]: I have made a pull request for this change in `sora-sdk`.
After it will be approved I will fix it in this PR

File: src/main/java/jp/co/soramitsu/sora/didresolver/controllers/DIDResolverController.java [hunk 6]
```diff
@@ -61,29 +84,39 @@ public void deleteDDO(
...
+
+    log.debug("proof has been successfully verified for DDO with DID {}", ddo.getId());
+  }
+
+  private void checkCreatorValidity(DDO ddo) {
+    if (isNull(ddo.getProof())) {
+      throw new InvalidProofException(ddo.getId().toString());
+    }
+    DID proofCreator = ddo.getProof().getOptions().getCreator();
+    if (!verifyService.isCreatorInAuth(proofCreator, ddo.getAuthentication())
```
[the remark this answers is not in the document]
Comment (JIexa) [reply src/main/java/jp/co/soramitsu/sora/didresolver/controllers/DIDResolverController.java]: it is a good question.
@Warchant  can you please clarify "do we need to check whether the **creator** is in the **Authentication** section

File: src/main/java/jp/co/soramitsu/sora/didresolver/services/impl/VerifyServiceImpl.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (tdorzhi) [reply src/main/java/jp/co/soramitsu/sora/didresolver/services/impl/VerifyServiceImpl.java]: pls refer to it here, so we see the status

File: src/main/java/jp/co/soramitsu/sora/didresolver/services/VerifyService.java (hunk 4 above)
Comment (tdorzhi) [reply src/main/java/jp/co/soramitsu/sora/didresolver/services/VerifyService.java]: also, void methods often are considered as ones that produce some side-effect, such as interaction with I/O (database, filesystem, http, stdout) or that modify in-place passed object (setters, or some collection methods like `List#add`), with that said, I consider changing to boolean return value

File: build.gradle [hunk 7]
```diff
@@ -44,20 +44,19 @@ ext {
...
     springSwaggerVersion = '2.9.2'
     commonsLangVersion = '3.8'
-//    log4jVersion = '2.11.1'
-//    slf4jVersion = '1.7.25'
     aspectjVersion = '1.9.1'
     spongycastleVersion = '1.58.0.0'
     mockitoCoreVersion = '2.22.0'
     hibernateTypesVersion = '2.3.0'
     testContainersVersion = '1.8.3'
-    postgresVersion = '42.2.4'
```
Comment (Warchant) [inline build.gradle]: why did you remove postgres?

File: build.gradle [hunk 8]
```diff
@@ -67,70 +66,53 @@ repositories {
...
-            "org.springframework.boot:spring-boot-starter-web",
-            "org.springframework.boot:spring-boot-starter-validation",
-            "org.springframework.boot:spring-boot-starter-data-jpa"
-    ].each { dep ->
-        compile(dep) {
-            exclude module: "spring-boot-starter-logging"
-            exclude module: "logback-classic"
-        }
-    }*/
+    compile "com.github.soramitsu:sora-sdk:${soraSDKVersion}"
```
Comment (Warchant) [inline build.gradle]: update sora-sdk version

File: build.gradle [hunk 9]
```diff
@@ -67,70 +66,53 @@ repositories {
...
 
-    //commons
     compile("org.apache.commons:commons-lang3:${commonsLangVersion}")
 
     //DB
-    // Handy library which allows to map json/jsonb columns to java type on in entity column!!
-    // https://github.com/vladmihalcea/hibernate-types
     compile("com.vladmihalcea:hibernate-types-52:${hibernateTypesVersion}")
-    compile("org.postgresql:postgresql:${postgresVersion}")
+    compile("org.postgresql:postgresql")
```
Comment (Warchant) [inline build.gradle]: set version

File: src/main/java/jp/co/soramitsu/sora/didresolver/commons/DIDTypeEnum.java (hunk 2 above)
Comment (Warchant) [reply src/main/java/jp/co/soramitsu/sora/didresolver/commons/DIDTypeEnum.java]: if we have any constraints on username in account service, then they need to be implemented here also

File: src/main/java/jp/co/soramitsu/sora/didresolver/validation/DIDValidator.java [hunk 10]
```diff
@@ -11,8 +11,8 @@
 public class DIDValidator implements ConstraintValidator<DIDConstraint, String> {
```
[the remark this answers is not in the document]
Comment (Warchant) [reply src/main/java/jp/co/soramitsu/sora/didresolver/validation/DIDValidator.java]: +

This class can be removed.

File: build.gradle (hunk 7 above)
Comment (JIexa) [reply build.gradle]: it is managed by spring-boot-gradle-plugin

File: src/test/java/jp/co/soramitsu/sora/didresolver/controllers/DIDResolverControllerTest.java [hunk 11]
```diff
@@ -1,18 +1,23 @@
...
 import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
 
 import java.time.Instant;
-import java.util.Optional;
 import jp.co.soramitsu.sora.didresolver.exceptions.DIDNotFoundException;
+import jp.co.soramitsu.sora.sdk.did.model.dto.DID;
+import jp.co.soramitsu.sora.sdk.did.parser.generated.ParserException;
 import org.junit.Test;
```
Comment (rogachevsn55) [inline src/test/java/jp/co/soramitsu/sora/didresolver/controllers/DIDResolverControllerTest.java]: You are removed dependency to junit vintage engine, but didn't change tests to junit 5. How it works right now?

File: src/test/java/jp/co/soramitsu/sora/service/DIDServiceImplTest.java [hunk 12]
```diff
@@ -63,27 +64,19 @@ public void setUp() {
...
   @Test
   public void givenCorrectIrohaIdWhenCalledReadAssertAccountEntityReturned()
-      throws UnparseableException {
-    val did = "did:sora:iroha:vasya@home.ru";
+      throws UnparseableException, ParserException {
+
+    val did = "did:sora:soraUser8";
     val ddo = new DDO();
-    ddo.setId(did);
+    ddo.setId(DID.randomUUID());
```
Comment (rogachevsn55) [inline src/test/java/jp/co/soramitsu/sora/service/DIDServiceImplTest.java]: static import

File: src/test/java/jp/co/soramitsu/sora/util/DataProvider.java [hunk 13]
```diff
@@ -6,74 +6,78 @@
...
-  public static final String DID = "did:sora:uuid:caab4570-5f3f-4050-8d61-15306dea4bcf";
+  public static final String TEST_DID = "did:sora:soraUser8";
 
-  public static final String ID_BASE = DID + "#keys-";
+  public static final String ID_BASE = TEST_DID + "#keys-";
 
-  public static final byte[] KEY_VALUE = DatatypeConverter
-      .parseHexBinary(
+  public static final byte[] KEY_VALUE =
+      DatatypeConverter.parseHexBinary(
```
Comment (rogachevsn55) [inline src/test/java/jp/co/soramitsu/sora/util/DataProvider.java]: static import

File: src/test/java/jp/co/soramitsu/sora/util/DataProvider.java [hunk 14]
```diff
@@ -6,74 +6,78 @@
...
     return publicKeys;
   }
 
   public Proof getProofForTest() {
-    return new Proof(getCryptoType(CryptoActionTypeEnum.SIGNATURE),
-        Instant.now().truncatedTo(ChronoUnit.SECONDS),
-        URI.create(ID_BASE + random.nextInt(KEYS_COUNT)), KEY_VALUE, null, null);
+    try {
+      Options options = new Options(
+          SignatureTypeEnum.Ed25519Sha3Signature, Instant.now(),
```
Comment (rogachevsn55) [inline src/test/java/jp/co/soramitsu/sora/util/DataProvider.java]: static import

File: src/test/java/jp/co/soramitsu/sora/util/DataProvider.java [hunk 15]
```diff
@@ -6,74 +6,78 @@
...
+          DID.parse(ID_BASE + 2),
```
Comment (rogachevsn55) [inline src/test/java/jp/co/soramitsu/sora/util/DataProvider.java]: and here

File: src/test/java/jp/co/soramitsu/sora/didresolver/controllers/DIDResolverControllerTest.java (hunk 11 above)
Comment (JIexa) [reply src/test/java/jp/co/soramitsu/sora/didresolver/controllers/DIDResolverControllerTest.java]: junit4 derives from springboot-starter-test and Test class does not use `Runner` of junit5

**Comment by Warchant** [comment]
https://github.com/soramitsu/did-resolver/blob/859ff532122b56ab6ec854e4eb3b0a7ba7ad9415/build.gradle#L124-L126

This is the reason why CI fails. You enabled only html output. You need also to enable xml test output.
